File: python/src/wdl_model/model/expressions/wdl_map_literal.py
from __future__ import annotations

from collections import deque
from dataclasses import dataclass, field

from wdl_model.model.base import WdlExpresionKeyValue

from .wdl_expression import ComponentType, WdlExpression


class WdlMapEntry(WdlExpresionKeyValue):
    pass


@dataclass
class WdlMapLiteral(WdlExpression):
    _entries: deque[WdlMapEntry] = field(default_factory=deque)

    def entries(self) -> deque[WdlMapEntry]:
        return self._entries

    def __str__(self) -> str:
        return (
            "{"
            + ", ".join(f"{e.getKey()}: {e.getValue()}" for e in self._entries)
            + "}"
        )

    def componentType(self) -> ComponentType:
        return ComponentType.MAP_LITERAL

File: java/src/main/java/com/myriad/wdl/model/expressions/WdlMapLiteral.java
package com.myriad.wdl.model.expressions;

import com.myriad.wdl.model.base.WdlKeyValue.WdlExpresionKeyValue;
import java.util.ArrayDeque;

public final class WdlMapLiteral implements WdlExpression {
  private final ArrayDeque<WdlMapEntry> entries = new ArrayDeque<>();

  public WdlMapLiteral() {}

  public ArrayDeque<WdlMapEntry> entries() {
    return entries;
  }

  @Override
  public String toString() {
    return getClass().getSimpleName();
  }

  @Override
  public ComponentType componentType() {
    return ComponentType.MAP_LIT;
  }

  public static class WdlMapEntry extends WdlExpresionKeyValue {
    public WdlMapEntry() {
      super();
    }

    public WdlMapEntry(WdlExpression key, WdlExpression value) {
      super(key, value);
    }

    public WdlMapEntry(WdlExpression key) {
      super(key);
    }
  }
}
